### PR TITLE
Update all unique flasks to reflect current version

### PR DESCRIPTION
**Pull request by QuickStick123**
### Description of the problem being solved:
Several flask mods are out of date and not supported and updated on the items
This updates ranges so the desirable option is on the right hand side of the range
This add support for them along with a couple other easy changes I thought to do when looking at this stuff.

### CODE REVIEW

File: src/Data/Uniques/flask.lua [hunk 1]
```diff
@@ -417,36 +402,33 @@ Variant: Current (Proliferation)
...
+{variant:16,17,18}(150-125)% increased Charges per use
+Shocks nearby Enemies during Effect, causing 10% increased Damage taken
+You are Shocked during Effect, causing 50% increased Damage taken
 {variant:1,5,11}Damage Penetrates 10% Lightning Resistance during Effect
 {variant:16}Damage Penetrates 6% Lightning Resistance during Effect
 {variant:2,6,9}Adds (15-25) to (70-90) Lightning Damage to Spells during Effect
 {variant:12}Adds (25-35) to (110-130) Lightning Damage to Spells during Effect
 {variant:17}Adds (10-15) to (55-65) Lightning Damage to Spells during Effect
-{variant:3,7,13}Adds (25-35) to (110-130) Lightning Damage to Attacks during Effect
+{variant:3,7,13}Adds (25-35) to (110-130) Lightning Damage to Spells during Effect
```
Comment (Wires77) [inline src/Data/Uniques/flask.lua:415]: This should still be `Attacks` for the pre-3.15.0 attack version of the flask

File: src/Modules/CalcOffence.lua [hunk 2]
```diff
@@ -2695,9 +2695,9 @@ function calcs.offence(env, actor, activeSkill)
 			output.EnergyShieldOnHit = 0
 			output.ManaOnHit = 0
 		else
-			output.LifeOnHit = skillModList:Sum("BASE", cfg, "LifeOnHit") + enemyDB:Sum("BASE", cfg, "SelfLifeOnHit")
-			output.EnergyShieldOnHit = skillModList:Sum("BASE", cfg, "EnergyShieldOnHit") + enemyDB:Sum("BASE", cfg, "SelfEnergyShieldOnHit")
-			output.ManaOnHit = skillModList:Sum("BASE", cfg, "ManaOnHit") + enemyDB:Sum("BASE", cfg, "SelfManaOnHit")
+			output.LifeOnHit = not skillModList:Flag(cfg, "CannotLeechLife") and (skillModList:Sum("BASE", cfg, "LifeOnHit") + enemyDB:Sum("BASE", cfg, "SelfLifeOnHit")) or 0
+			output.EnergyShieldOnHit = not skillModList:Flag(cfg, "CannotLeechEnergyShield") and (skillModList:Sum("BASE", cfg, "EnergyShieldOnHit") + enemyDB:Sum("BASE", cfg, "SelfEnergyShieldOnHit")) or 0
+			output.ManaOnHit = not skillModList:Flag(cfg, "CannotLeechMana") and (skillModList:Sum("BASE", cfg, "ManaOnHit") + enemyDB:Sum("BASE", cfg, "SelfManaOnHit")) or 0
```
Comment (Wires77) [inline src/Modules/CalcOffence.lua:2700]: "On hit" effects should still apply if you can't leech.  Where did this and the other change stem from?

Edit: I see the ModParser change for "Cannot gain Life", but that is still a different flag from "CannotLeechLife"

File: src/Modules/CalcOffence.lua [hunk 3]
```diff
@@ -2709,9 +2709,9 @@ function calcs.offence(env, actor, activeSkill)
 			output.EnergyShieldOnKill = 0
 			output.ManaOnKill = 0
 		else
-			output.LifeOnKill = m_floor(skillModList:Sum("BASE", cfg, "LifeOnKill"))
-			output.EnergyShieldOnKill = m_floor(skillModList:Sum("BASE", cfg, "EnergyShieldOnKill"))
-			output.ManaOnKill = m_floor(skillModList:Sum("BASE", cfg, "ManaOnKill"))
+			output.LifeOnKill = not skillModList:Flag(cfg, "CannotLeechLife") and (m_floor(skillModList:Sum("BASE", cfg, "LifeOnKill"))) or 0
```
Comment (Wires77) [inline src/Modules/CalcOffence.lua:2712]: This is the other leech change.

**Review by Wires77 — CHANGES_REQUESTED** [review]
Most of this looks good, but there are a few things that should be changed.

File: src/Data/Uniques/flask.lua [hunk 4]
```diff
@@ -203,133 +195,130 @@ Variant: Pre 3.0.0
 Variant: Pre 3.15.0
 Variant: Pre 3.16.0
 Variant: Current
-Requires Level 68
-{variant:2}(-10-10)% increased Charges used
-{variant:3,4}(125-150)% increased Charges used
-{variant:3}(40-60)% reduced duration
-{variant:4}(40-60)% less duration
+LevelReq: 68
+{variant:2}(-10-10)% reduced Charges per use
```
[the remark this answers is not in the document]
Comment (QuickStick123) [reply src/Data/Uniques/flask.lua:199]: I can change them back to that and just flip the ranges if you like.

File: src/Modules/CalcOffence.lua (hunk 2 above)
Comment (QuickStick123) [reply src/Modules/CalcOffence.lua]: I updated it to be special and just be a cannot gain flag

**Review by Wires77 — APPROVED** [review]

**Pull request merged**